source/puppet/4.4/reference/static_catalogs.md

```diff
@@ -0,0 +1,130 @@
+
+If static catalogs are enabled but none of the relevant Puppet Server settings are configured, the `code_id` parameter defaults to a null value and the `static_file_content` API endpoint always returns the latest content. To populate the `code_id` with a more useful identifier and have the `static_file_content` endpoint return the appropriate version of the file's content, you must specify scripts or commands that provide Puppet with the right information.
+
+Puppet Server locates these commands via the `code-id-command` and `code-content-command` settings in Puppet Server's [`puppetserver.conf`][] file. Puppet Server runs the `code-id-command` each time it compiles a static catalog, and it runs the `code-content-command` each time an agent requests file contents from the `static_file_content` endpoint.
+
+> **Puppet Enterprise note:** You can set `code-id-command` and `code-content-command` in the PE console by changing the `code_id_command` and `code_content_command` parameters. Note that these only take effect if [file sync][] is disabled by setting the `file_sync_enabled` console parameter to false.
+
+Puppet Server validates the standard output of each of these scripts, and if the output's acceptable, it adds the results to the catalog as their respective parameters' values. This lets you use any versioning or synchronization tools you want, as long as you can write scripts that produce a valid hash for the `code_id` and code content using the catalog's `code_id` and file's environment.
+
+The `code-id-command` and `code-content-command` scripts can be as simple or complex as necessary. For instance, if files in an environment are simply managed by git, a `code-id-command` script can be nothing more than this, with the environment name passed to the script as the first parameter:
+
+``` bash
+#!/bin/bash
+set -e
+if [[ -z $1 ]]; then
+  echo Expected an environment >&2
+  exit 1
+fi
+cd /etc/puppetlabs/code/environments/$1 && git rev-parse HEAD
+```
+
+As long as the script's standard output is a valid hash, Puppet Server will store it as the catalog's `code_id`. If the script returns a non-zero exit code, Puppet Server logs an error.
+
+A `code-content-command` script can also be simple. The following example assumes the environment name is passed to the script as the first parameter, the `code_id` is passed as the second parameter, and the path to the file resource from its `content_uri` is passed as the third parameter:
```

Comment:
Instead of "The following example assumes ...", I'd probably phrase this as something more like "Code content scripts will get the environment name passed as the first command-line argument, ..."